extensions/wikia/PageHeader/Subtitle.class.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+namespace Wikia\PageHeader;
+
+use FakeSkin;
+use Html;
+use RequestContext;
+
+class Subtitle {
+	/**
+	 * @var string
+	 */
+	public $subtitle;
+
+	/**
+	 * @var string
+	 */
+	public $pageSubtitle;
+
+	/**
+	 * @var mixed|null
+	 */
+	public $suppressPageSubtitle;
+
+	/**
+	 * @var \SkinTemplate
+	 */
+	private $skinTemplate;
+
+	/**
+	 * @var bool
+	 */
+	private static $onEditPage = false;
+
+	public function __construct( \WikiaApp $app ) {
+		$this->suppressPageSubtitle = $app->wg->SuppressPageSubtitle;
+		$this->request = RequestContext::getMain()->getRequest();
+		$this->skinTemplate = $app->getSkinTemplateObj();
+		if ( !$this->suppressPageSubtitle ) {
+			$this->subtitle = $this->getSubtitle();
+			//watch list uses that, pageSubject?
+			$this->pageSubtitle = $this->getPageSubtitle();
+		}
+	}
+
+	/**
+	 * Detect we're on edit (or diff) page
+	 */
+	public static function isEditPage() {
+		$wg = \F::app()->wg;
+		return !empty( Hooks::$onEditPage ) ||
+		       !is_null( $wg->Request->getVal( 'diff' ) ) /* diff pages - RT #69931 */ ||
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)